### PR TITLE
Add handling for websocket disconnection.

### PR DESCRIPTION
**Pull request by apexskier**
Workaround fix for #1. Based on https://github.com/slackhq/python-slackclient/issues/118#issuecomment-250025739

Note: I did this in github's editor, so no linting or testing has been done. The most likely problem is that WebSocketConnectionClosedException needs to be imported from somewhere else.

### CODE REVIEW

**Comment by reillysiemens** [comment]
@apexskier: I truly appreciate that you took the time to do this. I can't merge this until I've tested it, but I'll give it a look when I've got a spare moment. Thanks!

File: wb2k/__main__.py [hunk 1]
```diff
@@ -110,12 +110,20 @@ def cli(channel, verbose):
 
         logger.info("Listening for joins in #{}".format(channel))
         while True:
-
-            # Handle dem messages!
-            for message in sc.rtm_read():
-                handle_message(message, channel, channel_id, sc, logger)
-
-            time.sleep(0.5)
+            # handle bug in slack websocket connection https://github.com/reillysiemens/wb2k/issues/1
```
Comment (reillysiemens) [inline wb2k/__main__.py]: I don't think we need to keep this comment.

File: wb2k/__main__.py [hunk 2]
```diff
@@ -110,12 +110,20 @@ def cli(channel, verbose):
...
+            try:
+                # Handle dem messages!
+                for message in sc.rtm_read():
+                    handle_message(message, channel, channel_id, sc, logger)
+
+                time.sleep(0.5)
+            except WebSocketConnectionClosedException:
```
Comment (reillysiemens) [inline wb2k/__main__.py]: Does this exception not have to be imported from some library?

**Comment by reillysiemens** [comment]
@apexskier: Overall I think this is quite good and it appears to have improved the connectivity of `wb2k`, but just mindlessly attempting to reconnect to Slack doesn't seem like the most responsible thing to me. Perhaps some form of exponential back off might be good? After a certain number of failures `wb2k` could justifiably throw its hands up and bail.

File: wb2k/__main__.py [hunk 3]
```diff
@@ -109,13 +111,35 @@ def cli(channel, verbose):
...
+                    time.sleep((backoff ** 2) / 4)
+                else:
+                    logger.info("Reconnected to Slack")
+
+                retry_count += 1
 
     else:
         sys.exit(bail('fatal', 'red', "Couldn't connect to Slack"))
+
+if __name__ == '__main__':
```
Comment (apexskier) [inline wb2k/__main__.py]: This allows you to clone the repo, then run `python wb2k` and have everything start.

**Comment by reillysiemens** [comment]
@apexskier: Thanks for adding the exponential backoff. LGTM! :ship:

**Pull request merged**